src/Prototype/Types.hs

```diff
@@ -49,17 +56,31 @@ instance FromJSON TodoState
 --------------------------------------------------------------------------------
 data Operation = BumpCounter
 
+newtype Namespace = Namespace { _unNamespace :: NonEmptyText }
```

Comment:
- I think we should enforce something like only alphanumeric characters and dashes, can't start with a dash, no more than one contiguous dash, can't end with a dash.
- We should also have a denylist to avoid users registering "useful" routes, e.g. "blog"
- Do you think we should have a specialized `Text` wrapper ?